### PR TITLE
separation startup scripts and application config from yaml files

### PR DESCRIPTION
**Pull request by onlymellb**

### CODE REVIEW

File: charts/tidb-cluster/templates/_helpers.tpl [hunk 1]
```diff
@@ -14,3 +14,11 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 {{- $name := default .Chart.Name .Values.nameOverride -}}
 {{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
+
+{{- define "tidb-cluster.utils.template" -}}
```
Comment (gregwebs) [inline charts/tidb-cluster/templates/_helpers.tpl]: I would like to give the template helpers generic names if they are not specific to our chart. This makes it easier to re-use them in different charts.

Comment (tennix) [reply charts/tidb-cluster/templates/_helpers.tpl]: Maybe `chart.configmap-template`?

Comment (onlymellb) [reply charts/tidb-cluster/templates/_helpers.tpl]: Maybe use `helm-toolkit.utils.template` instead @gregwebs @tennix ?

**Comment by onlymellb** [comment]
The pump/drainer startup script has been separated.

File: charts/tidb-cluster/templates/_helpers.tpl (hunk 1 above)
Comment (tennix) [reply charts/tidb-cluster/templates/_helpers.tpl]: ok, lgtm.

**Comment by tennix** [comment]
For the configuration files, I think it's better to add the filename suffix for toml, yaml format.

**Comment by tennix** [comment]
I've run `helm template charts/tidb-cluster --name=demo --namespace=tidb` and got the following errors:
```
Error: render error in "tidb-cluster/templates/tikv-configmap.yaml": template: tidb-cluster/templates/tikv-configmap.yaml:13:42: executing "tidb-cluster/templates/tikv-configmap.yaml" at <include "helm-toolki...>: error calling include: template: no template "helm-toolkit.utils.template" associated with template "gotpl"
```
Have you tested the templates?

**Pull request merged**